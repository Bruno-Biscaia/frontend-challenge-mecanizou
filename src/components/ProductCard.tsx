import Link from 'next/link';

type Props = {
  id: string;
  name: string;
  price: number;
  image: string;
};

export default function ProductCard({ id, name, price, image }: Props) {
  return (
    <Link
      href={`/products/${id}`}
      className="block border rounded hover:shadow p-4"
    >
      <img
        src={image}
        alt={name}
        className="h-40 w-full object-cover mb-2 rounded"
      />
      <h3 className="font-semibold">{name}</h3>
      <p>R$ {price.toFixed(2)}</p>
    </Link>
  );
}
