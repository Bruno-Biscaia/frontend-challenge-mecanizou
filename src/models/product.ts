export type Product = {
  id: string;
  name: string;
  subTitle: string;
  description: string;
  price: number;
  image: string;
  numberOfReviews: number;
  reviews: number;
  categories: string;
};
