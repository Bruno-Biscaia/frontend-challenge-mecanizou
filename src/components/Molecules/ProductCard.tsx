// src/components/ProductCard.tsx
import Image from 'next/image';
import Link from 'next/link';
import { Product } from '@/models/product';
import { Typography } from '@/components/Atoms/Typography';

interface Props {
  product: Product;
}

export function ProductCard({ product }: Props) {
  return (
    <Link href={`/products/${product.id}`} className="group">
      <div>
        <Image
          alt={product.subTitle}
          src={product.image}
          width={500}
          height={500}
          className="aspect-square w-full rounded-lg  object-cover group-hover:opacity-75 xl:aspect-7/8"
        />

        <Typography as="h3" className="mt-4 text-sm ">
          {product.name}
        </Typography>

        <Typography as="p" className="mt-1 text-lg font-medium ">
          R$ {product.price.toFixed(2)}
        </Typography>
      </div>
    </Link>
  );
}
