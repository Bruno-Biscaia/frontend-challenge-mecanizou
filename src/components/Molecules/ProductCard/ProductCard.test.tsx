import { render, screen } from '@testing-library/react';
import { ProductCard } from './ProductCard';
import { Product } from '@/models/product';

const productMock: Product = {
  id: '1',
  name: 'Tênis Esportivo',
  price: 199.9,
  image:
    'https://images.unsplash.com/photo-1752384876830-1291984ec767?q=80&w=709&auto=format&fit=crop&ixlib=rb-4.1.0&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D',
  reviews: 4,
  numberOfReviews: 10,
  subTitle: 'Ótimo para corrida',
  description: 'está é a descrição do produto',
  categories: 'sports',
};

describe('ProductCard', () => {
  it('renderiza nome, preço e imagem', () => {
    render(<ProductCard product={productMock} />);
    expect(screen.getByText(productMock.name)).toBeInTheDocument();
    expect(
      screen.getByText(`R$ ${productMock.price.toFixed(2)}`)
    ).toBeInTheDocument();
    const img = screen.getByRole('img') as HTMLImageElement;
    expect(img).toHaveAttribute('src', productMock.image);
  });

  it('mantém o mesmo snapshot', () => {
    const { container } = render(<ProductCard product={productMock} />);
    expect(container).toMatchSnapshot();
  });
});
