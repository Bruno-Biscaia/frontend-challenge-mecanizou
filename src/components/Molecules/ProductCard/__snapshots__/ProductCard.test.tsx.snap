// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ProductCard mantém o mesmo snapshot 1`] = `
<div>
  <a
    class="group"
    data-testid="product-card"
    href="/produtos/1"
  >
    <div>
      <img
        alt="Ótimo para corrida"
        class="aspect-square w-full rounded-lg  object-cover group-hover:opacity-75 xl:aspect-7/8"
        height="500"
        src="https://images.unsplash.com/photo-1752384876830-1291984ec767?q=80&w=709&auto=format&fit=crop&ixlib=rb-4.1.0&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D"
        width="500"
      />
      <h3
        class="text-base leading-relaxed mt-4 text-sm "
      >
        Tênis Esportivo
      </h3>
      <p
        class="text-base leading-relaxed mt-1 text-lg font-medium "
      >
        R$ 
        199.90
      </p>
    </div>
  </a>
</div>
`;
